src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -186,42 +186,44 @@ static bool IsAccessibleFromGenBinders(ITypeSymbol type)
                     RegisterTypeDiagnostic(type, diag);
                 }
 
-                if (spec is null)
-                {
-                    return null;
-                }
-
-                string @namespace = spec.Namespace;
-                if (@namespace is not null and not "<global namespace>")
+                if (spec is { Namespace: string @namespace } && @namespace is not "<global namespace>")
                 {
                     _sourceGenSpec.Namespaces.Add(@namespace);
                 }
 
                 return _createdSpecs[type] = spec;
             }
 
-            private void RegisterTypeForBindCoreMainGen(TypeSpec typeSpec)
+            private bool TryRegisterTypeForBindCoreMainGen(ComplexTypeSpec type)
             {
-                if (typeSpec.NeedsMemberBinding)
+                if (type.HasBindableMembers)
                 {
-                    RegisterTypeForMethodGen(MethodsToGen_CoreBindingHelper.BindCoreMain, typeSpec);
-                    RegisterTypeForBindCoreGen(typeSpec);
-                    _sourceGenSpec.MethodsToGen_CoreBindingHelper |= MethodsToGen_CoreBindingHelper.AsConfigWithChildren;
+                    bool registeredForBindCoreGen = TryRegisterTypeForBindCoreGen(type);
+                    Debug.Assert(registeredForBindCoreGen);
+
+                    RegisterTypeForMethodGen(MethodsToGen_CoreBindingHelper.BindCoreMain, type);
+                    Register_AsConfigWithChildren_HelperForGen_IfRequired(type);
+                    return true;
                 }
+
+                return false;
             }
 
-            private void RegisterTypeForBindCoreGen(TypeSpec typeSpec)
+            private bool TryRegisterTypeForBindCoreGen(ComplexTypeSpec type)
             {
-                if (typeSpec.NeedsMemberBinding)
+                if (type.HasBindableMembers)
                 {
-                    RegisterTypeForMethodGen(MethodsToGen_CoreBindingHelper.BindCore, typeSpec);
+                    RegisterTypeForMethodGen(MethodsToGen_CoreBindingHelper.BindCore, type);
+                    return true;
                 }
+
+                return true;
```

Comment:
Yes thanks. The difference after I correct this is

- if the type appears as a member in the graph, we'd skip binding for that property
- if it's a root config type, we'd still generate an interceptor, but it will be a no op (e.g. `public static void Bind_EmptyType(this IConfig, object) => { }`)

I'll add a test.